cli/commands/run/cli.go

```diff
@@ -96,3 +102,50 @@ func validateCommand(opts *options.TerragruntOptions) error {
 func isTerraformPath(opts *options.TerragruntOptions) bool {
 	return strings.HasSuffix(opts.TerraformPath, options.TerraformDefaultPath)
 }
+
+// wrapWithStackGenerate wraps a CLI command to automatically generate stack configurations
+// when running with --all or --graph flags, unless --no-stack-generate is specified.
+// This allows users to transparently use stack features without manual stack generation.
+func wrapWithStackGenerate(opts *options.TerragruntOptions, cmd *cli.Command) *cli.Command {
+	// Wrap the original command action with our stack generation logic
+	cmd = cmd.WrapAction(func(ctx *cli.Context, action cli.ActionFunc) error {
+		// Set the default stack configuration path in the working directory
+		stackConfigPath := filepath.Join(opts.WorkingDir, config.DefaultStackFile)
+		opts.TerragruntStackConfigPath = stackConfigPath
```

Comment:
I think TerragruntStackConfigPath should be set when `shouldGenerateStack` is true